### PR TITLE
Test more upgrade scenarios

### PR DESCRIPTION
**Pull request by sbodagala**
# Description

This PR tests the following upgrade scenarios:

- Upgrading a cluster when no remote storage processes are restarted
- Upgrading a cluster with a pending pod

## Type of change

- Other (testing related)

## Discussion

*Are there any design details that you would like to discuss further?*

No.

## Testing

*Please describe the tests that you ran to verify your changes. Unit tests?
Manual testing?*

Manual testing (ran the tests locally).

*Do we need to perform additional testing once this is merged, or perform in a larger testing environment?*

These tests will need to be run as part of regression testing.

## Documentation

*Did you update relevant documentation within this repository?*

*If this change is adding new functionality, do we need to describe it in our user manual?*

*If this change is adding or removing subreconcilers, have we updated the core technical design doc to reflect that?*

*If this change is adding new safety checks or new potential failure modes, have we documented and how to debug potential issues?*

N/A

## Follow-up

*Are there any follow-up issues that we should pursue in the future?*

No.

*Does this introduce new defaults that we should re-evaluate in the future?*

No.

### CODE REVIEW

File: e2e/test_operator_upgrades/operator_upgrades_test.go [hunk 1]
```diff
@@ -897,4 +897,17 @@ var _ = Describe("Operator Upgrades", Label("e2e"), func() {
 		fixtures.GenerateUpgradeTableEntries(testOptions),
 	)
 
+	FDescribeTable(
```
Comment (johscheuer) [inline e2e/test_operator_upgrades/operator_upgrades_test.go:900]:
```suggestion
	DescribeTable(
```

File: e2e/test_operator_ha_upgrades/operator_ha_upgrade_test.go [hunk 2]
```diff
@@ -389,4 +389,49 @@ var _ = Describe("Operator HA Upgrades", Label("e2e"), func() {
...
+
+			log.Println(
+				"Selected Pods:",
+				ignoreDuringRestart,
+				"to be skipped during the restart",
+			)
+			fdbCluster.GetRemote().SetIgnoreDuringRestart(ignoreDuringRestart)
+
+			// The cluster should still be able to upgrade.
+			Expect(fdbCluster.UpgradeCluster(targetVersion, true)).NotTo(HaveOccurred())
```
Comment (johscheuer) [inline e2e/test_operator_ha_upgrades/operator_ha_upgrade_test.go:431]: So the upgrades goes well without any issues?

File: e2e/test_operator_upgrades/operator_upgrades_test.go (hunk 1 above)
Comment (sbodagala) [reply e2e/test_operator_upgrades/operator_upgrades_test.go:900]: Yes, I missed this.

**Review by johscheuer — APPROVED** [review]
LGTM 👍

**Comment by johscheuer** [comment]
Lets wait for the result of the e2e test.

**Review by johscheuer — APPROVED** [review]
Thanks!

**Pull request merged**